src/Storages/StoragePostgreSQL.cpp

```diff
@@ -456,15 +458,24 @@ SinkToStoragePtr StoragePostgreSQL::write(
     return std::make_shared<PostgreSQLSink>(metadata_snapshot, pool->get(), remote_table_name, remote_table_schema, on_conflict);
 }
 
-StoragePostgreSQL::Configuration StoragePostgreSQL::processNamedCollectionResult(const NamedCollection & named_collection, bool require_table)
+StoragePostgreSQL::Configuration StoragePostgreSQL::processNamedCollectionResult(const NamedCollection & named_collection, PostgreSQLSettings& storage_settings, bool require_table)
 {
     StoragePostgreSQL::Configuration configuration;
+
+    ValidateKeysMultiset<ExternalDatabaseEqualKeysSet> optional_arguments = {"replace_query", "on_duplicate_clause", "addresses_expr", "host", "hostname", "port"};
+    auto postgresql_settings = storage_settings.all();
+    for (const auto & setting : postgresql_settings)
+    {
+        const auto & setting_name = setting.getName();
+        if (named_collection.has(setting_name))
+            storage_settings.set(setting_name, named_collection.get<String>(setting_name));
```

Comment:
In https://github.com/ClickHouse/ClickHouse/pull/52711#discussion_r1282946971 I meant not really this.
The code you removed was correct, there was no need to remove it. It was correct, but it was not full, there are two parts:
1. Adding args for validation: https://github.com/ClickHouse/ClickHouse/blob/6b330e30b025291e4a114d6ba19472eea658680c/src/Storages/StorageMySQL.cpp#L262-L265
2. Making use of the values: https://github.com/ClickHouse/ClickHouse/blob/6b330e30b025291e4a114d6ba19472eea658680c/src/Storages/StorageMySQL.cpp#L294-L299
So you had (1) and not (2). Now you have (2) but (1) is removed.